src/librustc/ty/inhabitedness/def_id_forest.rs

```diff
@@ -110,7 +110,7 @@ impl<'a, 'gcx, 'tcx> DefIdForest {
             where I: IntoIterator<Item=DefIdForest>
     {
         let mut ret = DefIdForest::empty();
-        let mut next_ret = SmallVec::new();
+        let mut next_ret: SmallVec<[DefId; 1]> = SmallVec::new();
```

Comment:
This feels like an unrelated change? Unless inference is failing, as I suspect it is, in which case I'm not sure we'd be able to land this -- too much breakage.